src/server/cookies.ts

```diff
@@ -31,6 +32,7 @@ export async function encrypt(
 
   const encryptedCookie = await new jose.EncryptJWT(payload)
     .setProtectedHeader({ enc: ENC, alg: ALG, ...additionalHeaders })
+    .setExpirationTime(expiration)
```

Comment:
This is the main thing this PR is fixing. The rest is just accommodating for this change.